### PR TITLE
Change handling of suites to stop prepending some test suite names to tests

### PR DESCRIPTION
**Pull request by stbenjam**
[TRT-1238](https://issues.redhat.com//browse/TRT-1238)

This fixes Sippy's handling of suites.  It makes it more consistent with how we're doing things in bigquery and component readiness, meaning we'll be able to bigquery consume data more easily: for example, we'll be able to sync from the junit table instead of directly from prow, or use the component mapping table in Sippy.

This PR:
 - limits the suites we import to the ones in our DB (this list is in sippy's code, but will probably eventually move to bigquery)
 - stop prepending suite names to the front of tests
 - has a script to remove the prefixes and update existing records

Merging suggestion:
- disable fetchdata and db backups, scale down sippy temporarily
- merge this PR
- run `go run ./scripts/updatesuites.go` (takes about ~4 hours)
- do a vacuum full (~2.5 hours)
- enable fetchdata and scale sippy back up

### CODE REVIEW

File: pkg/db/migrations.go [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+	"gorm.io/gorm"
+
+	"github.com/openshift/sippy/pkg/db/models"
+)
+
+// testNameWithoutSuite removes test suite prefixes from tests in the database
+// and assigns the suite to all the prow_job_run_tests. When the test can't be
+// renamed because the unprefixed version exists in the DB, use that one and remove
+// the prefixed version.
+func testNameWithoutSuite(dbc *DB) error {
```
Comment (stbenjam) [inline pkg/db/migrations.go:17]: Maybe this should just be a one-time script we run against the production DB?

**Comment by stbenjam** [comment]
/hold

Not ready to merge yet

File: scripts/updatesuites.go [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+					return err
+				}
+			} else { //nolint
+				log.Infof("existing test found, making it the default and removing the old one...")
+				err := dbc.Transaction(func(tx *gorm.DB) error {
+					// Update rows in the prow_job_run_tests table and then delete the old oldTest row.
+					if err := tx.Model(&models.ProwJobRunTest{}).Where("test_id = ?", oldTest.ID).Updates(models.ProwJobRunTest{TestID: newTest.ID, SuiteID: &suiteID}).Error; err != nil {
+						log.WithError(err).Warningf("error updating prow_job_run_tests for oldTest ID %d", oldTest.ID)
+						return err
+					}
```
Comment (dgoodwin) [inline scripts/updatesuites.go:124]: Can you log the number of rows affected easily here?

**Review by dgoodwin — COMMENTED** [review]
Really nice work on a very delicate operation. Could you adjust the logging a little per this review and then post a log of migrating the dev backup? Would love to eye ball what it's going to do.

File: scripts/updatesuites.go (hunk 2 above)
Comment (stbenjam) [reply scripts/updatesuites.go:135]: Yup, added.  14,838,319 was what I got for the snapshot restore

File: pkg/db/suites.go [hunk 3]
```diff
@@ -1,27 +1,52 @@
...
+	"Operator results",
+	"Symptom detection",
+	"Tests Suite",
+	"cluster install",
+	"cluster nodes ready",
+	"cluster nodes",
+	"gather core dump",
+	"hypershift-e2e",
+	"metal infra",
+	"step graph",
```
[the remark this answers is not in the document]
Comment (stbenjam) [reply pkg/db/suites.go:45]: I added a note to the development.md doc but I think this is transitional anyway.  I want to move this to bigquery and sync it down to sippy so we have a canonical list of suites TRT cares about

**Pull request merged**